xarray/core/computation.py

```diff
@@ -619,9 +619,10 @@ def _apply_with_dask_atop(func, args, input_dims, output_dims, signature,
 
     (out_ind,) = output_dims
     # skip leading dimensions that we did not insert with broadcast_compat_data
-    atop_args = [element
-                 for (arg, dims) in zip(args, input_dims)
-                 for element in (arg, dims[-getattr(arg, 'ndim', 0):])]
+    atop_args = [
+        element
+        for (arg, dims) in zip(args, input_dims)
+        for element in (arg, dims[-getattr(arg, 'ndim', 0) or len(dims):])]
```

Comment:
So the change here is that you don't include any of the dims when `arg.ndim == 0` (or ndim does exist)? That seems fine. 

Style comment. This double nested list comprehension is pretty tough to read. There's just a lot going on. I might suggest rewriting it so its easier to read. 

```Python
atop_args = []
for arg, dims in zip(args, input_dims):
    s = -getattr(arg, 'ndim', 0) or len(dims)
    for element in (arg, dims[s:]):
        atop_args.append(element)
```

Reply:
Yes, the goal is to get the last `ndim` items from `dims`.

I'll rewrite this with the loop, that would indeed be clearer.